src/utilities/game_launcher.py

```diff
@@ -48,8 +48,19 @@ def launch_runelite_with_settings(bot, settings_file: Path):
     key = bot.game_title.lower()
     EXECPATH = data.get(key, "")
 
+    if not os.path.exists(EXECPATH) and sys.platform == 'darwin':
```

Comment:
I'm sure `subprocess.Popen()` would be able to execute and run the RuneLite game. Is the issue that it is not appearing in Finder? I believe this can be solved by adding the `.app` extension to the list of searchable files in the `__locate_executable()` function below.

My Mac is kind of reserved for work so I don't know if I want to tinker with RS stuff, but maybe you could give that a shot?

Reply:
Nope, woudln't work, just re-tried it. When one selects an app, even if the filter is '*' it still opens `/Applications/RuneLite.app/` as an executable instead of opening it as a folder. There's no way I found to go all the way up to `/Applications/RuneLite.app/Contents/MacOS/RuneLite` just using the "open" dialog window.

Reply:
Could you just rename `/Applications/RuneLite.app/` to `/Applications/RuneLite/`? Or do folders in the Applications folder need to have that extension?

Reply:
If I rename it in the cache file, it just opens the file opener dialog.
Otherwise it just throws an exception if I choose runelite from the dialog:
![Screenshot 2023-01-08 at 08 49 52](https://user-images.githubusercontent.com/480103/211185985-63260dcb-dc37-4885-99c1-934c892539b1.png)
<img width="950" alt="Screenshot 2023-01-08 at 08 54 42" src="https://user-images.githubusercontent.com/480103/211186055-a6c11e4d-96e1-48ab-8d52-a178d86605aa.png">

Reply:
What about adding this code to the `__locate_executables()` function?
```python
def __locate_executable():
    """
    Opens a file dialog to allow the user to locate the game executable.
    """
    root = tk.Tk()
    root.withdraw()
    file_path = filedialog.askopenfilename(
        title="Select game executable file", filetypes=[("exe files", "*.exe"), ("AppImage files", "*.AppImage"), ("Java files", "*.jar")]
    )
    try:
        if not file_path:
            root.destroy()
            return None
        if platform.system() == "Darwin":
            file_path += "/Contents/MacOS/RuneLite"  # <-- Add this line
        file_path = Path(file_path)
    except TypeError:
        root.destroy()
        return None
    path_str = str(file_path)
    root.destroy()
    return path_str
```

Reply:
Yeah, this actually worked! I updated the PR